sdk/monitor/monitor-query/src/internal/modelConverters.ts

```diff
@@ -422,12 +435,17 @@ export function convertBatchQueryResponseHelper(
   }
 }
 
-export function mapError(error?: GeneratedErrorInfo): ErrorInfo | undefined {
+export function mapError(error?: GeneratedErrorInfo): LogsErrorInfo | undefined {
   if (error) {
+    let innermostError = error;
+    while (innermostError.innerError) {
+      innermostError = innermostError.innerError;
+    }
+
     return {
-      ...error,
       name: "Error",
-      innerError: mapError(error.innerError)
+      code: error.code,
+      message: error.message + innermostError.message
```

Comment:
may be something like `message: error.message + + ' ' +  innermostError.message`